test/libyul/yulOptimizerTests/reasoningBasedSimplifier/addmod.yul

```diff
@@ -26,6 +26,9 @@
 //     let z := calldataload(64)
 //     let result := addmod(x, y, z)
 //     if 0 { }
-//     if and(and(lt(x, 1000), lt(y, 1000)), lt(z, 1000)) { if 1 { sstore(0, 9) } }
+//     if and(and(lt(x, 1000), lt(y, 1000)), lt(z, 1000))
```

Comment:
This is bad and should not happen :(

Reply:
In the encoding, `result = wrap(x + y) % z` while the condition below is `(x + y) % z`. With the restrictions in this line, z3 should be able to determine that the condition is `1`.

On the other hand, if we switch to an LP solver, it will probably not understand it anyway...

Reply:
We should at least add a comment in this test.

Reply:
Added a comment